### PR TITLE
[AVRO-1837] Add support for Date, Time and Timestamp Logical Type in python Avro

### PR DESCRIPTION
**Pull request by abrarsheikh**
Need to be reviewed by : @clin7 @justinmcunn @premsantosh

### CODE REVIEW

**Comment by rdblue** [comment]
@abrarsheikh, this looks like a great contribution. Are you interested in getting it into Avro?

**Comment by abrarsheikh** [comment]
@rdblue : yes i will be putting this out for review soon, have some improvements in terms of design that i need to do.

File: lang/py/src/avro/io.py [hunk 1]
```diff
@@ -232,6 +237,15 @@ def read_utf8(self):
     """
     return unicode(self.read_bytes(), "utf-8")
 
+  def read_date_from_int(self):
+    """
+    Decode python date object from int
+    int stores the number of days from
+    the unix epoch, 1 January 1970 (ISO calendar).
+    """
+    days_since_epoc = self.read_int()
```
Comment (clin7) [inline lang/py/src/avro/io.py]: nitpick: days_since_epoc/h/

File: lang/py/src/avro/schema.py [hunk 2]
```diff
@@ -312,6 +312,18 @@ def name_ref(self, names):
   namespace = property(lambda self: self.get_prop('namespace'))
   fullname = property(lambda self: self._fullname)
 
+# Logical type class
+class LogicalSchema(Schema):
```
Comment (clin7) [inline lang/py/src/avro/schema.py]: You might want to base on https://github.com/Yelp/avro/blob/trunk/lang/py/src/avro/schema.py#L322.

File: lang/py/src/avro/schema.py [hunk 3]
```diff
@@ -722,7 +748,14 @@ def make_avsc_object(json_data, names=None):
   if hasattr(json_data, 'get') and callable(json_data.get):
     type = json_data.get('type')
     other_props = get_other_props(json_data, SCHEMA_RESERVED_PROPS)
+    logical_type = None
+    if 'logicalType' in json_data:
+      logical_type = json_data.get('logicalType')
+      if logical_type not in ['date']:
+        raise SchemaParseException("Currently does not support %s logical type" % logical_type)
```
Comment (clin7) [inline lang/py/src/avro/schema.py]: I wonder if we should do this validation, according to the spec: "Language implementations must ignore unknown logical types when reading, and should use the underlying Avro type. If a logical type is invalid, for example a decimal with scale greater than its precision, then implementations should ignore the logical type and use the underlying Avro type." 

@justinmcunn @premsantosh @abrarsheikh : what do you think?

**Comment by andybryant** [comment]
why was this closed without merging?

**Comment by abrarsheikh** [comment]
@andybryant @rdblue opened a new pull request #207

**Pull request closed without merging**